### ISSUE TITLE
[Windows 11 & v8.5.1] The context menu needs restarting explorer to appear

### ISSUE THREAD

**Issue by donho** [notepad-plus-plus/nppShell#2]
After installing v8.5.1 under Widows 11, users have to restart Explorer or OS to make "Edit with Notepad++" exploer context menu appear.

Ref:  https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13399

**Comment by GurliGebis** [comment]
I'm 99,9% sure I know what the issue is.
When installing the 32bit version on Windows 11, which is 64bit, the msix is not installed.

But registering the dll on Windows 11 tries to register the sparse package (like it should), which will fail.

I will include it in the installer changes PR I do later, then it should "just works"

**Comment by chip33** [comment]
@GurliGebis @donho 
The new nppShell is reading the OS CurrentBuildNumber 

`[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows NT\CurrentVersion]
"CurrentBuildNumber"="22624"` 

and installing the new [Sparse Package (NppShell.msix)](https://github.com/notepad-plus-plus/nppShell) if the build number corresponds to Windows 11.
This leaves win 11 users who use the classic context menu with no Notepad++ context menu entry at all.
Please could we include an option on  installation to choose the legacy ContextMenu that is still installed for Windows 10 users in 8.5.1?
This approach is used by WinRar as a way of  enabling users to chose between the Sparse Package  for the modern context menu or the legacy if they are using the classic context menu in Win 11.
Thanks.

**Comment by chip33** [comment]
Hi, 
I posted about this with a solution for those using the legacy context menu on Win 11 [here](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13399#issuecomment-1483799531)

**Comment by GurliGebis** [comment]
@chip33 like mentioned there, there is a bug in the current release that prevents it from working.
My PR in there should fix it - I will test it once it is merged and let you know

**Comment by chip33** [comment]
> @chip33 like mentioned there, there is a bug in the current release that prevents it from working. My PR in there should fix it - I will test it once it is merged and let you know

I am using the 64bit version of notepad++

**Comment by donho** [comment]
@GurliGebis 
> When installing the 32bit version on Windows 11, which is 64bit, the msix is not installed.
But registering the dll on Windows 11 tries to register the sparse package (like it should), which will fail.

v8.5.1 32bits installer on Windows 11 works fine - just tested it and we need only to restart OS (or explorer) to make the entry appears.

This issue is rather for eliminating the "restarting explorer" procedure - it will be a (VERY) NICE TO HAVE enhancement.

**Comment by GurliGebis** [comment]
@donho ahh yes.
That issue is fixed in the next dll, so as soon as you merge it and build an installer, I'll test it 😃

**Comment by donho** [comment]
@GurliGebis 
> That issue is fixed in the next dll, so as soon as you merge it and build an installer, I'll test it 😃

By "That issue" you means "restarting Explorer to make Edit with Notepad++" appear, right?

**Comment by GurliGebis** [comment]
Yes @donho, just need a new installer with my PR against this repo, with a new built dll

**Comment by GurliGebis** [comment]
@donho this might be a good test case for TEST5 😀 agree?

**Comment by TetraTheta** [comment]
Not sure if this is right place to report problem with NppShell, but in my environment, 'Open with Notepad++' context menu doesn't appear even after PC restart.

I updated NPP from v8.5 to v8.5.1 and found out that the context menu was gone. So after reading this issue, I rebooted my PC. But the context menu is still missing.

I even tried to register the DLL manually by running this command in Command Prompt with admin priviledge but it failed.
```batch
regsvr32 NppModernShell.dll
```
The error message was about 'entry-point DLLRegisterserver was not found'.
This kind of workaround worked with Mp3Tag which uses similar approach with context menu (`.msix` and `.dll`) and I wonder if this is related to this issue or not.

If not, I'm willing to post new issue about this.

---

I'm not sure if this is related to using StartAllBack which brings back old context menu.  
But manually re-registering context menu of Mp3Tag worked even with old context menu so I think this is NppShell problem.

**Comment by GurliGebis** [comment]
@TetraTheta there are several issues with the 8.5.1 implementation. We are working on fixing it for the next release.

**Comment by GurliGebis** [comment]
@donho lets try with TEST9 and see if it works.
@TetraTheta can you try and see if it works with this test version? http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST09/

**Comment by TetraTheta** [comment]
@GurliGebis Tested with `npp.8.5.1.Installer.x64.exe` and unfortunately, it didn't work.

I tried:
- Install via overwriting : The context menu is still missing
- Logout and re-login : The context menu is still missing
- Uninstall and re-install : The context menu is still missing
- Reboot PC : The context menu is still missing

I don't see any DLL or MSIX file in NPP install destination. Are you sure NppShell is included in TEST09 setup file?

**Comment by GurliGebis** [comment]
@TetraTheta is it a modified Windows 11 installation? (like some services removed or disabled)

**Comment by GurliGebis** [comment]
@TetraTheta also, if you run this command in powershell, what does it give you?
`Get-AppxPackage -Name NotepadPlusPlus`

**Comment by TetraTheta** [comment]
@GurliGebis I use modified Windows 11, but that wouldn't cause problem with context menu because I didn't removed any services while doing that.  
Also, as I mentioned before, I had no problem with registering Mp3Tag's context menu both automatically (by installer) and manually with this modified Windows 11.

Here is the output of running `Get-AppxPackage -Name NotepadPlusPlus` command in PowerShell.
```plaintext
Name              : NotepadPlusPlus
Publisher         : CN="Notepad++", O="Notepad++", L=Saint Cloud, S=Ile-de-France, C=FR
Architecture      : Neutral
ResourceId        :
Version           : 1.0.0.0
PackageFullName   : NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6
InstallLocation   : C:\Program Files\WindowsApps\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6
IsFramework       : False
PackageFamilyName : NotepadPlusPlus_7njy0v32s6xk6
PublisherId       : 7njy0v32s6xk6
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Developer
Status            : Ok
```

**Comment by TetraTheta** [comment]
I also tested [TEST10](http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST10/), but that didn't register the context menu either.  
If I'm the only one who experiences this, I will just resort to the old way of adding context menus, which messes up the registry.

**Comment by GurliGebis** [comment]
@TetraTheta what kind of modifications have you done to it?
It is strange Mp3tag works, since they do it the same way as far as I know.

**Comment by TetraTheta** [comment]
@GurliGebis I just removed some unwanted stuff like Windows Maps, Windows Media Player, etc. I didn't touch the services because I don't know them well. That also applies to library stuff.

Yeah, it is very strange that Mp3Tag works but NPP doesn't.  
Mp3Tag seems to be closed-source, so I can't tell the difference between Mp3Tag's implementation and NppShell's, not to mention my inexperience with this kind of stuff.

**Comment by GurliGebis** [comment]
@TetraTheta not sure if it makes any difference, but can you try with http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST10/ ?

**Comment by TetraTheta** [comment]
@GurliGebis I've already tested it

> I also tested [TEST10](http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST10/), but that didn't register the context menu either.

Is there any difference between that time and now?

**Comment by GurliGebis** [comment]
Only from the installer point of view

**Comment by TetraTheta** [comment]
I re-downloaded the TEST10 installer just a moment ago and installed it.  
I restarted Windows Explorer process(`explorer.exe`) and no context menu present.  
I rebooted my PC and still no context menu present.

At this point, I'm really not sure if it's just me who has this problem or if other people are also experiencing it.

**Comment by GurliGebis** [comment]
@TetraTheta can I get you to try using the Windows Sandbox feature, and installing it inside there.
It is a lightwigth, buildin VM based on your installation, so it would be interresting to see if the issue is also in there.

**Comment by TetraTheta** [comment]
@GurliGebis I could get the context menu inside Windows Sandbox both 'modern context menu' and 'classic context menu'.  
It was my problem that the context menu didn't showed up. Thanks for the help!

![1](https://user-images.githubusercontent.com/37061801/229092099-16f62ed7-b2e2-40bb-a4c6-6386536113c8.jpg)
![2](https://user-images.githubusercontent.com/37061801/229092108-23d3b776-6e07-468f-a0aa-9bf574cdfde2.jpg)

**Comment by donho** [comment]
@TetraTheta 
So could you confirm that TEST10 doesn't work for you due to your special configuration?
Could you elaborate more about your configuration?

**Comment by chip33** [comment]
I have a fairly new install of win 11 on a virtual machine, I will check it there and let you know :)

**Comment by donho** [comment]
@TetraTheta 
Thank you very much for your very detail information!

**Comment by GurliGebis** [comment]
@donho can you mark this issue as closed?
